content/courses/advanced-ada/parts/resource_management/anonymous_access_types.rst

```diff
@@ -252,21 +254,553 @@ cover more specific details on anonymous access types.
+
+Because the pool selection is hidden from us, this makes the memory
+deallocation more complicated. For example, we cannot instantiate the
+:ada:`Ada.Unchecked_Deallocation` procedure for anonymous access types. Also,
+some of the methods we could use to circumvent this limitation are error-prone,
+as we discuss in this section.
+
+Also, storage-related features are available: specifying the storage size |mdash|
+especially, specifying that the access type has a storage size of zero |mdash|
+isn't possible.
+
+Missing features
+^^^^^^^^^^^^^^^^
+
+Let's see a code example that shows some of the features that aren't available
+for anonymous access-to-object types:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.Missing_Anonymous_Access_To_Object_Features
+
+    with Ada.Unchecked_Deallocation;
+
+    package Missing_Features is
+
+       --  We cannot specify which pool will be used
+       --  in the anonymous access-to-object
+       --  allocation; the pool is selected by the
+       --  compiler:
+       IA : access Integer := new Integer;
+
+       --
+       --  All the features below aren't available
+       --  for an anonymous access-to-object:
+       --
+
+       --  Having a specific storage pool associated
+       --  with the access type:
+       type Integer_Access is
+          access Integer;
+
+       --  Specifying a deallocation function for the
+       --  access type:
+       procedure Free is
+         new Ada.Unchecked_Deallocation
+           (Object => Integer,
+            Name   => Integer_Access);
+
+       --  Specifying a limited storage size for
+       --  the access type:
+       type Integer_Access_Store_128 is
+          access Integer
+            with Storage_Size => 128;
+
+       --  Limiting the storage size for the
+       --  access type to zero:
+       type Integer_Access_Store_0 is
+          access Integer
+            with Storage_Size => 0;
+
+    end Missing_Features;
+
+In the :ada:`Missing_Features` package, we see some of the features that we
+cannot use for the anonymous :ada:`access Integer` type, but that are available
+for equivalent named access types:
+
+- There's no specific memory pool associated with the access object :ada:`IA`.
+  In contrast the :ada:`Integer_Access` has an associated pool.
+
+- We cannot instantiate the :ada:`Ada.Unchecked_Deallocation` procedure for
+  the :ada:`access Integer` type. However, we can instantiate it for named
+  access types such as the :ada:`Integer_Access` type.
+
+- We cannot use the :ada:`Storage_Size` attribute for the :ada:`access Integer`
+  type, but we're allowed to use it with named access types, which we do in the
+  declaration of the :ada:`Integer_Access_Store_128` and
+  :ada:`Integer_Access_Store_0` types.
+
+
+Dangerous memory deallocation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We could make up for the absence of the :ada:`Ada.Unchecked_Deallocation`
```

Comment:
Since this approach should not be done, let me suggest that "We could make up" be changed to "We might think that we could make up"

Reply:
Yes, that's much better, thanks for the suggestion!